### PR TITLE
Dcp 456 states

### PR DESCRIPTION
**Pull request by jacobwindsor**
dcp-546

- Formats and shows the new submission states for graph validation (removing need for `graphValidationState`
- Allows graph validation whenever submission is `METADATA_VALID`
- Refactors a bit to move all of the submission states to enums

### CODE REVIEW

File: src/app/project/project.component.ts [hunk 1]
```diff
@@ -200,7 +200,7 @@ export class ProjectComponent implements OnInit {
   canSubmit(project: Project) {
     return this.userIsWrangler &&
       !project.hasOpenSubmission &&
-      project.validationState.toUpperCase() !== 'INVALID' &&
+      project.validationState.toUpperCase() !== 'METADATA INVALID' &&
```
Comment (MightyAx) [inline src/app/project/project.component.ts:203]: Could you use the constant you created here?
```
import {SUBMISSION_STATES} from "@shared/constants";
```

File: src/app/projects/services/project-cache.service.spec.ts [hunk 2]
```diff
@@ -35,7 +35,7 @@ describe('ProjectCacheService', () => {
       uuid: {uuid: 'test123'},
       events: [],
       dcpVersion: '123',
-      validationState: 'VALID',
+      validationState: 'METADATA VALID',
```
Comment (MightyAx) [inline src/app/projects/services/project-cache.service.spec.ts:38]: Could you use the constant you created here too?

File: src/app/submission/components/metadata-list/metadata-list.component.html [hunk 3]
```diff
@@ -69,20 +69,18 @@
     <!--Row Detail Template -->
     <ngx-datatable-row-detail [rowHeight]="'auto'" #myDetailRow>
       <ng-template let-rowIndex="rowIndex" let-row="row" let-expanded="expanded" ngx-datatable-row-detail-template>
-        <ng-container *ngIf="row['validationState'] === 'Invalid' || row['validationState'] === 'Valid'">
+        <ng-container *ngIf="row['validationState'] === 'Metadata invalid' || row['validationState'] === 'Metadata valid'">
```
Comment (MightyAx) [inline src/app/submission/components/metadata-list/metadata-list.component.html:72]: Could you use the constant you created here as well?

File: src/app/submission/components/metadata-list/metadata-list.component.html [hunk 4]
```diff
@@ -69,20 +69,18 @@
...
           <h3>Metadata Validation Report</h3>
           <!--Metadata validation-->
           <div style="padding-left:35px;" *ngIf="row['validationState'] === 'Invalid'">
             <ng-container *ngFor="let error of getValidationErrors(row)">
               <span class="vf-u-text-color--red" title="{{error}}">{{error}}</span><br>
             </ng-container>
           </div>
-          <div style="padding-left:35px;" *ngIf="row['validationState'] === 'Valid'">
+          <div style="padding-left:35px;" *ngIf="row['validationState'] === 'Metadata valid'">
```
Comment (MightyAx) [inline src/app/submission/components/metadata-list/metadata-list.component.html:80]: Additionally, Could you use the constant you created here?

File: src/app/submission/components/validation-summary/validation-summary.component.html [hunk 5]
```diff
@@ -10,16 +10,10 @@
         [summaries]="[
           {
             count: summary.invalidBiomaterials,
-            textOne: 'metadata error',
-            textMany: 'metadata errors',
+            textOne: 'error',
+            textMany: 'errors',
```
Comment (MightyAx) [inline src/app/submission/components/validation-summary/validation-summary.component.html:14]: Is it relevant to use the constant you created throughout this files references to `'error'` and `'errors'`?

**Review by MightyAx — CHANGES_REQUESTED** [review]
Really like the new `SUBMISSION_STATES` constant.
There are a few places where it could be used that are missing.

Could you search and replace all relevant usages of these strings?

File: src/app/submission/components/validation-summary/validation-summary.component.html (hunk 5 above)
Comment (jacobwindsor) [reply src/app/submission/components/validation-summary/validation-summary.component.html:14]: You mean use an enum for "error" and "errors"

Comment (MightyAx) [reply src/app/submission/components/validation-summary/validation-summary.component.html:14]: I think my pattern matching was a bit off, I thought `'error'` and `'errors'` were submission states.
If they aren't then this is fine as it is.

**Review by MightyAx — APPROVED** [review]
Thanks for the changes

File: src/app/shared/components/submission-state/submission-state.component.ts [hunk 6]
```diff
@@ -19,38 +25,33 @@ export class SubmissionStateComponent {
...
-    if (submissionState === 'Processing'
-      || submissionState === 'Cleanup'
-      || submissionState === 'Archiving'
-      || submissionState === 'Exporting') {
-      return 'warning-invert';
-    }
-
-    if (submissionState === 'Complete' || submissionState === 'Exported' || submissionState === 'Archived') {
-      return 'success-invert';
+    switch(submissionState) {
```
Comment (amnonkhen) [inline src/app/shared/components/submission-state/submission-state.component.ts:28]: Is there a reason not to use a map?

Comment (jacobwindsor) [reply src/app/shared/components/submission-state/submission-state.component.ts:28]: I think this is easier to reason about than a map

**Review by aaclan-ebi — APPROVED** [review]

**Pull request merged**